src/components/pages/strategies/Active/StrategyTab.tsx

```diff
@@ -1,30 +1,135 @@
+  align-items: baseline;
+`;
+
+// TODO: move to constants
+const TEN_DECIMAL = new Decimal("10");
+
+// TODO: move to utils
+function calculatePriceFromPartial(
+  price: Decimal,
+  denominatorDecimals: number,
+  numeratorDecimals: number
+): Decimal {
+  return price.mul(TEN_DECIMAL.pow(denominatorDecimals - numeratorDecimals));
+}
+
+function formatBrackets(strategy: Strategy): BracketRowData[] {
+  const {
+    baseTokenDetails: { decimals: baseTokenDecimals },
+    quoteTokenDetails: { decimals: quoteTokenDecimals },
+    brackets,
+    prices,
+    tokenBaseBalances,
+    tokenQuoteBalances,
+  } = strategy;
+
+  return brackets.map((bracket, index) => ({
+    lowPrice: calculatePriceFromPartial(
+      prices[index * 2],
+      baseTokenDecimals,
+      quoteTokenDecimals
+    ),
+    highPrice: calculatePriceFromPartial(
+      prices[index * 2 + 1],
+      baseTokenDecimals,
+      quoteTokenDecimals
+    ),
+    balanceBase: new Decimal(
+      formatAmount(tokenBaseBalances[bracket.address], baseTokenDecimals)
+    ),
+    balanceQuote: new Decimal(
+      formatAmount(tokenQuoteBalances[bracket.address], quoteTokenDecimals)
+    ),
+  }));
+}
+
 export const StrategyTab = memo(function StrategyTab(
   props: Props
 ): JSX.Element {
   const { strategy } = props;
+  const {
+    baseTokenDetails,
+    baseTokenAddress,
+    quoteTokenDetails,
+    quoteTokenAddress,
+    brackets,
+    priceRange,
+  } = strategy;
+
+  const { price } = useGetPrice({
+    source: "GnosisProtocol",
+    baseToken: baseTokenDetails,
+    quoteToken: quoteTokenDetails,
+  });
+
+  const {
+    baseTokenBrackets,
+    quoteTokenBrackets,
+  } = calculateBracketsFromMarketPrice({
+    marketPrice: price || ZERO_DECIMAL,
+    totalBrackets: brackets.length,
+    lowestPrice: priceRange?.lower || ZERO_DECIMAL,
+    highestPrice: priceRange?.upper || ZERO_DECIMAL,
+  });
+
+  const tableBrackets = formatBrackets(strategy);
 
   return (
     <Wrapper>
       <BracketsViewer
         type="strategy"
-        baseTokenAddress={strategy.baseTokenAddress}
-        quoteTokenAddress={strategy.quoteTokenAddress}
-        lowestPrice={strategy.priceRange?.lower.toString()}
-        highestPrice={strategy.priceRange?.upper.toString()}
-        totalBrackets={strategy.brackets?.length}
+        baseTokenAddress={baseTokenAddress}
+        quoteTokenAddress={quoteTokenAddress}
+        lowestPrice={priceRange?.lower.toString()}
+        highestPrice={priceRange?.upper.toString()}
+        totalBrackets={brackets.length}
+        leftBrackets={baseTokenBrackets}
+        rightBrackets={quoteTokenBrackets}
+        startPrice={price?.isFinite() ? price.toString() : "N/A"}
       />
+      <Grid>
+        <BracketsTable
+          baseTokenAddress={baseTokenAddress}
+          quoteTokenAddress={quoteTokenAddress}
+          type="left"
+          brackets={tableBrackets.slice(0, baseTokenBrackets)}
+        />
+        <div style={{ justifySelf: "center" }}>TODO</div>
```

Comment:
This will cause rerenders. Move to variable outside function

Reply:
Memoized